src/components/DayPicker.jsx

```diff
@@ -316,8 +316,20 @@ class DayPicker extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { isFocused } = this.props;
-    const { focusedDate } = this.state;
+    const {
+      orientation, daySize, isFocused, numberOfMonths,
+    } = this.props;
+    const { focusedDate, monthTitleHeight } = this.state;
+
+    if (
+      this.isHorizontal()
+      && (orientation !== prevProps.orientation || daySize !== prevProps.daySize)
+    ) {
+      const visibleCalendarWeeks = this.calendarMonthWeeks.slice(0, numberOfMonths);
```

Comment:
ah shit, this should be `this.calendarMonthWeeks.slice(1, numberOfMonths + 1);` (we have to slice off the first and last months because they are hidden). I will update this.